modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/QueryTemplate.java

```diff
@@ -104,7 +103,7 @@ public ExecutionPlan map(PlanningContext ctx) {
     private List<Fragment> replace(List<Fragment> fragments, Fragment fragment, List<Fragment> replacement) {
         assert !nullOrEmpty(replacement);
 
-        Map<Long, Long> newTargets = new HashMap<>();
+        Long2LongOpenHashMap newTargets = new Long2LongOpenHashMap();
```

Comment:
we must revise all usages of new intMap, for example, few lines below we get Long and check in on null - it's wrong way.

Reply:
Revised all changes of Int and Long maps. In places with null check I replaced it with "containsKey" function.